Tests/Doctrine/Orm/Extension/PaginationExtensionTest.php

```diff
@@ -0,0 +1,214 @@
+        $this->requestStack->push(Request::create('/api/dummies', 'GET', [
+            'enablePagination' => 1,
+            'itemsPerPage' => 15,
+            'page' => 3,
+        ]));
+
+        $extension = new PaginationExtension($this->managerRegistry, $this->requestStack);
+        $extension->applyToCollection($this->resource, $this->queryBuilder);
+
+        $this->assertNull($this->queryBuilder->getFirstResult());
+        $this->assertNull($this->queryBuilder->getMaxResults());
+    }
+
+    public function testSupportsResult()
+    {
+        $this->requestStack->push(Request::create('/api/dummies', 'GET', [
+            'enablePagination' => 1,
+            'itemsPerPage' => 15,
+            'page' => 3,
+        ]));
+
+        $extension = new PaginationExtension($this->managerRegistry, $this->requestStack);
+        $this->assertTrue($extension->supportsResult($this->resource));
+    }
+
+    public function testSupportsResultNoRequest()
+    {
+        $extension = new PaginationExtension($this->managerRegistry, $this->requestStack);
+        $this->assertFalse($extension->supportsResult($this->resource));
+    }
+
+    public function testSupportsResultEmptyRequest()
+    {
+        $this->requestStack->push(Request::create('/api/dummies'));
+
+        $extension = new PaginationExtension($this->managerRegistry, $this->requestStack);
+        $this->assertFalse($extension->supportsResult($this->resource));
+    }
+
+    public function testSupportsResultClientNotAllowedToPaginate()
+    {
+        $this->resource->initClientAllowedToEnablePagination(false);
+        $this->requestStack->push(Request::create('/api/dummies', 'GET', [
+            'enablePagination' => 1,
+            'itemsPerPage' => 15,
+            'page' => 3,
+        ]));
+
+        $extension = new PaginationExtension($this->managerRegistry, $this->requestStack);
+        $this->assertFalse($extension->supportsResult($this->resource));
+    }
+
+    public function testSupportsResultPaginationDisabled()
+    {
+        $this->resource->initClientAllowedToEnablePagination(false);
+        $this->resource->initClientAllowedToChangeItemsPerPage(false);
+        $this->requestStack->push(Request::create('/api/dummies', 'GET', [
+            'enablePagination' => 1,
+            'itemsPerPage' => 15,
+            'page' => 3,
+        ]));
+
+        $extension = new PaginationExtension($this->managerRegistry, $this->requestStack);
+        $this->assertFalse($extension->supportsResult($this->resource));
+    }
+
+    public function testGetResult()
+    {
+        $this->requestStack->push(Request::create('/api/dummies', 'GET', [
+            'enablePagination' => 1,
+            'itemsPerPage' => 15,
+            'page' => 1,
+        ]));
+
+        $extension = new PaginationExtension($this->managerRegistry, $this->requestStack);
+        $paginator = $extension->getResult($this->queryBuilder);
```

Comment:
When I run test I usually delete the whole cache to avoid such problems with a command like `rm -Rf features/fixtures/TestApp/cache/* && vendor/bin/behat`